
<!DOCTYPE html>


<html lang="en" data-content_root="../../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>The Age of Embeddings: Representations Are the Real Output &#8212; Babu group AI journal club</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css?v=6644e6bb" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'content/ai_pi_book/part2/part2_chapter1_age_of_embeddings';</script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="Multi-Task Learning: One Model, Many Outputs" href="part2_chapter2_multi_task_learning.html" />
    <link rel="prev" title="Why Modern AI Feels Alien (Even for researchers With Statistical and ML Background)" href="part2_why_modern_ai_feels_alien.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../../_static/jc_logo.png" class="logo__image only-light" alt="Babu group AI journal club - Home"/>
    <script>document.write(`<img src="../../../_static/jc_logo.png" class="logo__image only-dark" alt="Babu group AI journal club - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../../intro.html">
                    Babu Group AI Journal Club
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../state_of_ai_for_biology.html">State of AI for Biology: A Comprehensive Overview</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../protein_book/protein_book.html">Protein Book</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter1_chemical_properties.html">Chapter 1: Chemical Properties of Polypeptides</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter2_biosynthesis.html">Chapter 2: Biosynthesis of Proteins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter3_evolutionary_origins.html">Chapter 3: Evolutionary and Genetic Origins of Protein Sequences</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter4_physical_interactions.html">Chapter 4: Physical Interactions That Determine the Properties of Proteins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter5_conformational_properties.html">Chapter 5: Conformational Properties of Polypeptide Chains</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter6_folded_conformations.html">Chapter 6: The Folded Conformations of Globular Proteins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter7_proteins_in_solution.html">Chapter 7: Proteins in Solution and in Membranes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter8_interactions_with_molecules.html">Chapter 8: Interactions with Other Molecules</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter9_enzyme_catalysis.html">Chapter 9: Enzyme Catalysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../protein_book/chapter10_degradation.html">Chapter 10: Degradation</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../gene_book/gene_book.html">Gene Book</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter1_genes_are_dna.html">Chapter 1: Genes are DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter2_interrupted_gene.html">Chapter 2: The Interrupted Gene</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter3_genome_content.html">Chapter 3: The Content of the Genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter4_clusters_repeats.html">Chapter 4: Clusters and Repeats</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter5_messenger_rna.html">Chapter 5: Messenger RNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter6_protein_synthesis.html">Chapter 6: Protein Synthesis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter7_genetic_code.html">Chapter 7: Using the Genetic Code</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter8_protein_localization.html">Chapter 8: Protein Localization</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter9_transcription.html">Chapter 9: Transcription</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter10_operon.html">Chapter 10: The Operon</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter11_regulatory_circuits.html">Chapter 11: Regulatory Circuits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter12_phage_strategies.html">Chapter 12: Phage Strategies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter13_the_replicon.html">Chapter 13: The Replicon</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter14_dna_replication.html">Chapter 14: DNA Replication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter15_recombination_and_repair.html">Chapter 15: Recombination and Repair</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter16_transposons.html">Chapter 16: Transposons</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter17_retroviruses_and_retroposons.html">Chapter 17: Retroviruses and Retroposons</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter18_dna_rearrangement.html">Chapter 18: Rearrangement of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter19_chromosomes.html">Chapter 19: Chromosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter20_nucleosomes.html">Chapter 20: Nucleosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter21_promoters_and_enhancers.html">Chapter 21: Promoters and Enhancers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter22_activating_transcription.html">Chapter 22: Activating Transcription</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter23_controlling_chromatin_structure.html">Chapter 23: Controlling Chromatin Structure</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter24_rna_splicing.html">Chapter 24: RNA Splicing and Processing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter25_catalytic_rna.html">Chapter 25: Catalytic RNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter26_immune_diversity.html">Chapter 26: Immune Diversity</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter27_protein_trafficking.html">Chapter 27: Protein Trafficking</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter28_signal_transduction.html">Chapter 28: Signal Transduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter29_cell_cycle_growth_regulation.html">Chapter 29: Cell Cycle Growth Regulation</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter30_oncogenes_and_cancer.html">Chapter 30: Oncogenes and Cancer</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../gene_book/chapter31_gradients__cascades__and_signaling_pathways.html">Chapter 31: Gradients, Cascades, and Signaling Pathways</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../cell_book/cell_book.html">Cell Book</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter1_cells_genomes.html">Chapter 1: Cells and Genomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter2_cell_chemistry_and_biosynthesis.html">Chapter 2: Cell Chemistry and Biosynthesis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter3_proteins.html">Chapter 3: Proteins</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter4_dna_and_chromosomes.html">Chapter 4: DNA and Chromosomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter5_dna_replication__repair__recombination.html">Chapter 5: DNA Replication, Repair, Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter6_genome_to_protein.html">Chapter 6: How Cells Read the Genome: From DNA to Protein</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter7_gene_expression_control.html">Chapter 7: Control of Gene Expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter8_manipulation_methods.html">Chapter 8: Manipulating Proteins, DNA, and RNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter9_visualizing_cells.html">Chapter 9: Visualizing Cells</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter10_membrane_structure.html">Chapter 10: Membrane Structure</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter11_membrane_transport_and_electrical_properties.html">Chapter 11: Membrane Transport and Electrical Properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter12_intracellular_compartments_and_protein_sorting.html">Chapter 12: Intracellular Compartments and Protein Sorting</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter13_intracellular_vesicular_traffic.html">Chapter 13: Intracellular Vesicular Traffic</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter14_energy_conversion__mitochondria_and_chloroplasts.html">Chapter 14: Energy Conversion: Mitochondria and Chloroplasts</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter15_cell_communication.html">Chapter 15: Cell Communication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter16_the_cytoskeleton.html">Chapter 16: The Cytoskeleton</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter17_the_cell_cycle_and_programmed_cell_death.html">Chapter 17: The Cell Cycle and Programmed Cell Death</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter18_the_mechanics_of_cell_division.html">Chapter 18: The Mechanics of Cell Division</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter19_cell_junctions.html">Chapter 19: Cell Junctions, Cell Adhesion, and the Extracellular Matrix</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter20_germ_cells_and_fertilization.html">Chapter 20: Germ Cells and Fertilization</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter21_development.html">Chapter 21: Development of Multicellular Organisms</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter22_histology__lives_and_deaths_of_cells_in_tissues.html">Chapter 22: Histology: Lives and Deaths of Cells in Tissues</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter23_cancer.html">Chapter 23: Cancer</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter24_immune_system.html">Chapter 24: The Adaptive Immune System</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cell_book/chapter25_pathogens.html">Chapter 25: Pathogens</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../neural_sci_book/neural_sci_book.html">Principles of Neural Science and AI</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../drug_discovery_book/drug_discovery_book.html">Drug Discovery &amp; Development Book</a></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../ai_pi_book.html">The AI Book for PI in Biology</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../orientation/orientation_what_is_a_model.html">Understanding “Models”: From Biology to Deep Learning</a></li>
<li class="toctree-l2"><a class="reference internal" href="../orientation/orientation_training_vs_inference_vs_deployment.html">Training vs. Inference vs. Deployment</a></li>
<li class="toctree-l2"><a class="reference internal" href="../orientation/orientation_ai_in_biology_vs_chatgpt_vs_alphafold.html">Are ChatGPT and Protein Modeling AI the Same?</a></li>
<li class="toctree-l2"><a class="reference internal" href="../orientation/orientation_biology_needs_asi_not_agi.html">AI in Biology IS ASI—Artificial Specific Intelligence, not AGI</a></li>
<li class="toctree-l2"><a class="reference internal" href="../orientation/orientation_cambrian_era.html">Working in AI’s Pre-Textbook, Cambrian Era</a></li>
<li class="toctree-l2"><a class="reference internal" href="../orientation/orientation_jargons_with_fixed_meanings.html">Jargons with Community-Agreed Fixed Meanings</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part1/part1_must_know_models.html">The Must-Know Models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part1/buzz_vs_reality.html">Trends, Buzzwords, and Reality: What These Concepts <em>Actually</em> Mean for Biology</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part1/part1_digest_a_paper.html">How to Digest a Paper when chatting with your choice of GPT</a></li>
<li class="toctree-l2"><a class="reference internal" href="part2_ai_fundamentals_intuition.html">AI Fundamentals at Intuition Level</a></li>
<li class="toctree-l2 current active has-children"><a class="reference internal" href="part2_why_modern_ai_feels_alien.html">Why Modern AI Feels Alien (Even for researchers With Statistical and ML Background)</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l3 current active"><a class="current reference internal" href="#">The Age of Embeddings: Representations Are the Real Output</a></li>
<li class="toctree-l3"><a class="reference internal" href="part2_chapter2_multi_task_learning.html">Multi-Task Learning: One Model, Many Outputs</a></li>
<li class="toctree-l3"><a class="reference internal" href="part2_chapter3_transfer_learning.html">Transfer Learning: Standing on the Shoulders of Giants</a></li>
<li class="toctree-l3"><a class="reference internal" href="part2_chapter4_new_functionalities.html">New Functionalities Beyond Classification</a></li>
<li class="toctree-l3"><a class="reference internal" href="part2_chapter5_data_representation.html">Data Representation: Tokenization Changes Everything</a></li>
</ul>
</details></li>
<li class="toctree-l2"><a class="reference internal" href="part2_emerging_subfields.html">Going Deeper: Less-Known but Promisingly Emerging Subfields</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part3/part3_browse_by_catalog.html">Browse by Catalog: Our Curated Collection of Tools</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part3/part3_design_your_own.html">When You Want to Design Your Own</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part4/part4_case_studies.html">Case Studies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part4/part4_protein_language_models.html">What Protein Language Models Actually Learn</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part4/part4_foundation_models_genomics.html">Actual Performance of Foundation Models in Genomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part4/part4_debates_clarifications.html">Debates and Clarifications</a></li>
<li class="toctree-l2"><a class="reference internal" href="../part4/part4_confused_subfields.html">Seemingly Confused Subfields Clarified</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../journal_club_book/journal_club_book.html">Journal Club Discussion Take-Away</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../journal_club_book/mind_maps_gallery.html">Mind Maps Gallery</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../journal_club_book/confusing_concepts.html">Frequently Confusing AI Concepts</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../journal_club_book/milestone_works.html">Most Important Milestone Works</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../journal_club_book/fundamental_concepts.html">Fundamental AI Concepts</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../journal_club_book/trend_shifts.html">Important Trend Shifts</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="../../selection_criteria.html">Selection Criteria</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ai_playground_access.html">Access to AI Playground</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contributors.html">Contributors</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fcontent/ai_pi_book/part2/part2_chapter1_age_of_embeddings.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../../_sources/content/ai_pi_book/part2/part2_chapter1_age_of_embeddings.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>The Age of Embeddings: Representations Are the Real Output</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-fundamental-paradigm-shift">The Fundamental Paradigm Shift</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#what-are-embeddings">What Are Embeddings?</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#key-properties-of-good-embeddings">Key Properties of Good Embeddings</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#from-hand-crafted-features-to-learned-representations">From Hand-Crafted Features to Learned Representations</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#the-traditional-approach">The Traditional Approach</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#the-embedding-revolution">The Embedding Revolution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#how-embeddings-are-learned">How Embeddings Are Learned</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#self-supervised-learning">Self-Supervised Learning</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#contrastive-learning">Contrastive Learning</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#why-embeddings-matter-for-biology">Why Embeddings Matter for Biology</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#transfer-learning">1. Transfer Learning</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-task-learning">2. Multi-Task Learning</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#similarity-search-and-clustering">3. Similarity Search and Clustering</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#interpretable-latent-spaces">4. Interpretable Latent Spaces</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#practical-considerations">Practical Considerations</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#embedding-quality-depends-on-training-data">Embedding Quality Depends on Training Data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#not-all-layers-are-equal">Not All Layers Are Equal</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#dimensionality-matters">Dimensionality Matters</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#common-pitfalls">Common Pitfalls</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#key-takeaways">Key Takeaways</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#further-reading">Further Reading</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="the-age-of-embeddings-representations-are-the-real-output">
<h1>The Age of Embeddings: Representations Are the Real Output<a class="headerlink" href="#the-age-of-embeddings-representations-are-the-real-output" title="Link to this heading">#</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>This page is incomplete and under active development.</strong> We welcome your suggestions and feedback! Please feel free to contribute or reach out with ideas for improvement.</p>
</div>
<section id="the-fundamental-paradigm-shift">
<h2>The Fundamental Paradigm Shift<a class="headerlink" href="#the-fundamental-paradigm-shift" title="Link to this heading">#</a></h2>
<p><strong>The Old View:</strong> A model’s job is to produce a final prediction (classification label, regression value, etc.)</p>
<p><strong>The New Reality:</strong> The intermediate representations (embeddings) are often more valuable than the final output.</p>
<p>Modern AI systems like language models, variational autoencoders (VAEs), and contrastive learning frameworks don’t just predict—they learn rich internal representations of data. For biology:</p>
<ul class="simple">
<li><p>A protein language model’s embedding captures evolutionary relationships, structural motifs, and functional properties</p></li>
<li><p>These embeddings can be used for downstream tasks the model was never explicitly trained for</p></li>
<li><p>Think of embeddings as a learned “language” for describing your data that captures patterns humans might never articulate</p></li>
</ul>
</section>
<hr class="docutils" />
<section id="what-are-embeddings">
<h2>What Are Embeddings?<a class="headerlink" href="#what-are-embeddings" title="Link to this heading">#</a></h2>
<figure class="align-center" id="embedding-concept">
<img alt="Conceptual illustration of embeddings" src="../../../_images/embedding_idea.png" />
<figcaption>
<p><span class="caption-number">Fig. 1 </span><span class="caption-text">The embedding concept: transforming discrete, high-dimensional, or complex data into continuous vector representations that capture semantic relationships.</span><a class="headerlink" href="#embedding-concept" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>At their core, <strong>embeddings</strong> are dense vector representations of data that encode meaningful relationships in a continuous space. Instead of representing data through sparse, hand-crafted features, embeddings allow models to learn compact representations where similar items are positioned close together.</p>
<figure class="align-center" id="embedding-vectors">
<a class="reference internal image-reference" href="../../../_images/embedding_illustration_1.png"><img alt="Vector representation of words in embedding space" src="../../../_images/embedding_illustration_1.png" style="width: 70%;" />
</a>
<figcaption>
<p><span class="caption-number">Fig. 2 </span><span class="caption-text">Words represented as dense vectors in continuous embedding space. Each word is mapped to a point where semantic relationships are preserved through geometric proximity.</span><a class="headerlink" href="#embedding-vectors" title="Link to this image">#</a></p>
</figcaption>
</figure>
<section id="key-properties-of-good-embeddings">
<h3>Key Properties of Good Embeddings<a class="headerlink" href="#key-properties-of-good-embeddings" title="Link to this heading">#</a></h3>
<ol class="arabic simple">
<li><p><strong>Dimensionality Reduction</strong>: Embeddings compress high-dimensional or complex data into lower-dimensional vectors (typically 128-1024 dimensions) while preserving the most important information.</p></li>
<li><p><strong>Semantic Similarity</strong>: Items that are functionally or structurally similar have embeddings that are close in vector space. Distance in embedding space corresponds to meaningful relationships.</p></li>
</ol>
<figure class="align-center" id="embedding-analogies">
<a class="reference internal image-reference" href="../../../_images/embedding_illustration_3.png"><img alt="Semantic relationships in embedding space showing king-queen-man-woman analogy" src="../../../_images/embedding_illustration_3.png" style="width: 80%;" />
</a>
<figcaption>
<p><span class="caption-number">Fig. 3 </span><span class="caption-text">Semantic relationships encoded as geometric relationships in embedding space. The famous word2vec analogy: “king” - “man” + “woman” ≈ “queen” demonstrates how embeddings capture meaningful semantic relationships through vector arithmetic.</span><a class="headerlink" href="#embedding-analogies" title="Link to this image">#</a></p>
</figcaption>
</figure>
<ol class="arabic simple" start="3">
<li><p><strong>Continuous Representation</strong>: Unlike discrete one-hot encodings, embeddings place data on a continuous manifold where interpolation is meaningful. Points between known examples represent plausible intermediate states.</p></li>
<li><p><strong>Task-Agnostic Features</strong>: Well-trained embeddings capture general properties of the data that are useful across many different downstream tasks, not just the original training objective.</p></li>
</ol>
</section>
</section>
<hr class="docutils" />
<section id="from-hand-crafted-features-to-learned-representations">
<h2>From Hand-Crafted Features to Learned Representations<a class="headerlink" href="#from-hand-crafted-features-to-learned-representations" title="Link to this heading">#</a></h2>
<section id="the-traditional-approach">
<h3>The Traditional Approach<a class="headerlink" href="#the-traditional-approach" title="Link to this heading">#</a></h3>
<p>Historically, machine learning required domain experts to manually engineer features:</p>
<ul class="simple">
<li><p>For proteins: amino acid composition, charge, hydrophobicity scores, secondary structure propensity</p></li>
<li><p>For images: edge detectors, color histograms, texture descriptors</p></li>
<li><p>For text: word counts, TF-IDF scores, hand-crafted linguistic features</p></li>
</ul>
<p><strong>Problems with this approach:</strong></p>
<ul class="simple">
<li><p>Labor-intensive and requires deep domain expertise</p></li>
<li><p>Features may not capture the patterns that matter for your specific task</p></li>
<li><p>Difficult to represent complex, hierarchical relationships</p></li>
<li><p>Fixed features can’t adapt as you get more data</p></li>
</ul>
</section>
<section id="the-embedding-revolution">
<h3>The Embedding Revolution<a class="headerlink" href="#the-embedding-revolution" title="Link to this heading">#</a></h3>
<p>Modern deep learning automatically learns representations through training:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Traditional approach: manual feature engineering</span>
<span class="n">protein_features</span> <span class="o">=</span> <span class="p">[</span>
    <span class="n">calculate_hydrophobicity</span><span class="p">(</span><span class="n">sequence</span><span class="p">),</span>
    <span class="n">count_charged_residues</span><span class="p">(</span><span class="n">sequence</span><span class="p">),</span>
    <span class="n">get_secondary_structure_propensity</span><span class="p">(</span><span class="n">sequence</span><span class="p">),</span>
    <span class="c1"># ... hundreds of hand-crafted features</span>
<span class="p">]</span>

<span class="c1"># Modern approach: learned embeddings</span>
<span class="n">embedding</span> <span class="o">=</span> <span class="n">protein_model</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="n">sequence</span><span class="p">)</span>  <span class="c1"># Returns 1024-dim vector</span>
<span class="c1"># The model learned what matters from millions of sequences</span>
</pre></div>
</div>
<figure class="align-center" id="embedding-transformation">
<a class="reference internal image-reference" href="../../../_images/Kernel-trick-By-transforming-the-original-space-left-into-a-space-of-increased.png"><img alt="Transformation from original space to higher-dimensional embedding space" src="../../../_images/Kernel-trick-By-transforming-the-original-space-left-into-a-space-of-increased.png" style="width: 75%;" />
</a>
<figcaption>
<p><span class="caption-number">Fig. 4 </span><span class="caption-text">The power of learned transformations: Data that is not linearly separable in the original space (left) can be transformed into a higher-dimensional embedding space (right) where patterns become clear and separable. Embeddings learn these transformations automatically from data rather than requiring hand-crafted feature engineering.</span><a class="headerlink" href="#embedding-transformation" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>The model discovers features that humans might never think to look for, often outperforming decades of domain knowledge encoded in hand-crafted features.</p>
</section>
</section>
<hr class="docutils" />
<section id="how-embeddings-are-learned">
<h2>How Embeddings Are Learned<a class="headerlink" href="#how-embeddings-are-learned" title="Link to this heading">#</a></h2>
<section id="self-supervised-learning">
<h3>Self-Supervised Learning<a class="headerlink" href="#self-supervised-learning" title="Link to this heading">#</a></h3>
<p>Many of the most powerful embedding models are trained through <strong>self-supervised learning</strong>—they learn from the data itself without requiring explicit labels:</p>
<p><strong>Protein Language Models (e.g., ESM, ProtBERT)</strong></p>
<ul class="simple">
<li><p>Training task: Predict masked amino acids in a sequence</p></li>
<li><p>What’s learned: Evolutionary constraints, structural compatibility, functional motifs</p></li>
<li><p>Result: Embeddings that capture “protein-ness” in general</p></li>
</ul>
<p><strong>Molecular Embeddings (e.g., MolCLR, GraphMVP)</strong></p>
<ul class="simple">
<li><p>Training task: Predict relationships between different views of molecules (2D graphs vs 3D conformations)</p></li>
<li><p>What’s learned: Chemical properties, reactivity patterns, binding characteristics</p></li>
<li><p>Result: Embeddings useful for drug discovery, property prediction, synthesis planning</p></li>
</ul>
</section>
<section id="contrastive-learning">
<h3>Contrastive Learning<a class="headerlink" href="#contrastive-learning" title="Link to this heading">#</a></h3>
<p>A particularly powerful approach for learning embeddings:</p>
<ol class="arabic simple">
<li><p><strong>Positive pairs</strong>: Create pairs of related items (same protein, different mutations; same molecule, different conformations)</p></li>
<li><p><strong>Negative pairs</strong>: Sample unrelated items</p></li>
<li><p><strong>Training objective</strong>: Push positive pairs closer together in embedding space while pushing negative pairs apart</p></li>
</ol>
<p>This creates embeddings where meaningful similarity corresponds to vector proximity.</p>
</section>
</section>
<hr class="docutils" />
<section id="why-embeddings-matter-for-biology">
<h2>Why Embeddings Matter for Biology<a class="headerlink" href="#why-embeddings-matter-for-biology" title="Link to this heading">#</a></h2>
<section id="transfer-learning">
<h3>1. Transfer Learning<a class="headerlink" href="#transfer-learning" title="Link to this heading">#</a></h3>
<p>Train once on massive datasets, use everywhere:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Use pre-trained protein embeddings</span>
<span class="n">esm_model</span> <span class="o">=</span> <span class="n">load_pretrained</span><span class="p">(</span><span class="s2">&quot;ESM-2&quot;</span><span class="p">)</span>  <span class="c1"># Trained on 250M sequences</span>

<span class="c1"># Your task: predict enzyme activity with only 500 labeled examples</span>
<span class="n">embeddings</span> <span class="o">=</span> <span class="n">esm_model</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="n">your_sequences</span><span class="p">)</span>
<span class="n">activity_predictor</span><span class="o">.</span><span class="n">train</span><span class="p">(</span><span class="n">embeddings</span><span class="p">,</span> <span class="n">your_labels</span><span class="p">)</span>
<span class="c1"># Often outperforms models trained from scratch on millions of examples</span>
</pre></div>
</div>
<p>The embeddings capture fundamental biology learned from vast sequence databases. You leverage this knowledge even with small datasets.</p>
</section>
<section id="multi-task-learning">
<h3>2. Multi-Task Learning<a class="headerlink" href="#multi-task-learning" title="Link to this heading">#</a></h3>
<p>One embedding, many predictions:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">embedding</span> <span class="o">=</span> <span class="n">protein_model</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="n">sequence</span><span class="p">)</span>

<span class="c1"># Use the same embedding for different tasks</span>
<span class="n">stability</span> <span class="o">=</span> <span class="n">stability_predictor</span><span class="p">(</span><span class="n">embedding</span><span class="p">)</span>
<span class="n">localization</span> <span class="o">=</span> <span class="n">localization_predictor</span><span class="p">(</span><span class="n">embedding</span><span class="p">)</span>
<span class="n">interaction_partners</span> <span class="o">=</span> <span class="n">interaction_predictor</span><span class="p">(</span><span class="n">embedding</span><span class="p">)</span>
</pre></div>
</div>
<p>The embedding serves as a universal representation that supports diverse biological questions.</p>
</section>
<section id="similarity-search-and-clustering">
<h3>3. Similarity Search and Clustering<a class="headerlink" href="#similarity-search-and-clustering" title="Link to this heading">#</a></h3>
<p>Find related proteins by embedding similarity:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Embed your query protein</span>
<span class="n">query_embedding</span> <span class="o">=</span> <span class="n">model</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="n">query_sequence</span><span class="p">)</span>

<span class="c1"># Search millions of proteins efficiently</span>
<span class="n">similar_proteins</span> <span class="o">=</span> <span class="n">embedding_database</span><span class="o">.</span><span class="n">nearest_neighbors</span><span class="p">(</span>
    <span class="n">query_embedding</span><span class="p">,</span> 
    <span class="n">k</span><span class="o">=</span><span class="mi">100</span>
<span class="p">)</span>
<span class="c1"># Returns functionally related proteins, even with low sequence identity</span>
</pre></div>
</div>
<p>This enables:</p>
<ul class="simple">
<li><p>Finding homologs that sequence alignment misses</p></li>
<li><p>Discovering proteins with similar functions but different evolutionary origins</p></li>
<li><p>Organizing the protein universe into functional families</p></li>
</ul>
</section>
<section id="interpretable-latent-spaces">
<h3>4. Interpretable Latent Spaces<a class="headerlink" href="#interpretable-latent-spaces" title="Link to this heading">#</a></h3>
<p>Well-trained embeddings often have interpretable structure:</p>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Explore what the model learned</span>
<span class="n">pca</span> <span class="o">=</span> <span class="n">PCA</span><span class="p">(</span><span class="n">n_components</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
<span class="n">embedding_2d</span> <span class="o">=</span> <span class="n">pca</span><span class="o">.</span><span class="n">fit_transform</span><span class="p">(</span><span class="n">embeddings</span><span class="p">)</span>

<span class="c1"># Visualize: proteins often cluster by:</span>
<span class="c1"># - Structural fold families</span>
<span class="c1"># - Functional categories  </span>
<span class="c1"># - Evolutionary relationships</span>
<span class="c1"># Even though the model was never told about these categories!</span>
</pre></div>
</div>
<figure class="align-center" id="embedding-space-viz">
<a class="reference internal image-reference" href="../../../_images/embedding_illustration_2.jpg"><img alt="Visualization of embedding space showing clusters and relationships" src="../../../_images/embedding_illustration_2.jpg" style="width: 85%;" />
</a>
<figcaption>
<p><span class="caption-number">Fig. 5 </span><span class="caption-text">Visualization of high-dimensional embeddings projected into 2D space. Similar items cluster together, revealing the learned structure and relationships. These patterns emerge naturally from training, without explicit supervision about categories or relationships.</span><a class="headerlink" href="#embedding-space-viz" title="Link to this image">#</a></p>
</figcaption>
</figure>
<p>The learned representations often rediscover biological organization principles.</p>
</section>
</section>
<hr class="docutils" />
<section id="practical-considerations">
<h2>Practical Considerations<a class="headerlink" href="#practical-considerations" title="Link to this heading">#</a></h2>
<section id="embedding-quality-depends-on-training-data">
<h3>Embedding Quality Depends on Training Data<a class="headerlink" href="#embedding-quality-depends-on-training-data" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Domain specificity</strong>: Protein embeddings from models trained on all proteins may not be optimal for your specific protein family</p></li>
<li><p><strong>Data scale</strong>: More training data generally produces better embeddings</p></li>
<li><p><strong>Task alignment</strong>: Embeddings from models trained on related tasks transfer better</p></li>
</ul>
</section>
<section id="not-all-layers-are-equal">
<h3>Not All Layers Are Equal<a class="headerlink" href="#not-all-layers-are-equal" title="Link to this heading">#</a></h3>
<p>Deep models create embeddings at multiple layers:</p>
<ul class="simple">
<li><p><strong>Early layers</strong>: Capture low-level features (local sequence patterns)</p></li>
<li><p><strong>Middle layers</strong>: Often contain the most generalizable representations</p></li>
<li><p><strong>Late layers</strong>: More specialized for the original training task</p></li>
</ul>
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="c1"># Many models let you choose which layer&#39;s embeddings to use</span>
<span class="n">embeddings_layer_6</span> <span class="o">=</span> <span class="n">model</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="n">sequence</span><span class="p">,</span> <span class="n">layer</span><span class="o">=</span><span class="mi">6</span><span class="p">)</span>   <span class="c1"># More general</span>
<span class="n">embeddings_layer_33</span> <span class="o">=</span> <span class="n">model</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="n">sequence</span><span class="p">,</span> <span class="n">layer</span><span class="o">=</span><span class="mi">33</span><span class="p">)</span>  <span class="c1"># More specialized</span>
</pre></div>
</div>
</section>
<section id="dimensionality-matters">
<h3>Dimensionality Matters<a class="headerlink" href="#dimensionality-matters" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><strong>Higher dimensions</strong>: Can capture more nuanced information but require more data and computation</p></li>
<li><p><strong>Lower dimensions</strong>: More efficient, less prone to overfitting, but may lose important details</p></li>
<li><p><strong>Typical ranges</strong>:</p>
<ul>
<li><p>Small molecules: 128-512 dimensions</p></li>
<li><p>Proteins: 512-1280 dimensions</p></li>
<li><p>Large biomolecular complexes: 1024-2048 dimensions</p></li>
</ul>
</li>
</ul>
</section>
</section>
<hr class="docutils" />
<section id="common-pitfalls">
<h2>Common Pitfalls<a class="headerlink" href="#common-pitfalls" title="Link to this heading">#</a></h2>
<ol class="arabic simple">
<li><p><strong>Assuming embeddings are magic</strong>: They’re only as good as their training. Embeddings from models trained on unrelated data may not help.</p></li>
<li><p><strong>Ignoring embedding space geometry</strong>: Different models use different distance metrics and normalization schemes. Always check the documentation.</p></li>
<li><p><strong>Freezing when you should fine-tune</strong>: Pre-trained embeddings are a starting point. For specialized tasks, fine-tuning the embedding model on your data often helps.</p></li>
<li><p><strong>Over-interpreting individual dimensions</strong>: While embeddings capture meaningful patterns, individual dimensions rarely correspond to interpretable features. The information is distributed across the entire vector.</p></li>
</ol>
</section>
<hr class="docutils" />
<section id="key-takeaways">
<h2>Key Takeaways<a class="headerlink" href="#key-takeaways" title="Link to this heading">#</a></h2>
<div class="tip admonition">
<p class="admonition-title">Remember</p>
<ol class="arabic simple">
<li><p><strong>Embeddings are the product</strong>: Modern AI’s real value often lies in its learned representations, not just its predictions</p></li>
<li><p><strong>Transfer learning at scale</strong>: Pre-trained embeddings let you leverage knowledge from massive datasets even when you have limited labeled data</p></li>
<li><p><strong>Similarity in embedding space is meaningful</strong>: Close vectors represent related biological entities—a powerful property for search, clustering, and analysis</p></li>
<li><p><strong>One representation, many uses</strong>: Good embeddings support diverse downstream tasks without modification</p></li>
<li><p><strong>Learn from the data</strong>: Embeddings often capture patterns and relationships that human experts never explicitly encoded</p></li>
</ol>
</div>
</section>
<hr class="docutils" />
<section id="further-reading">
<h2>Further Reading<a class="headerlink" href="#further-reading" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><span id="id1">[]</span> - ESM protein language models</p></li>
<li><p><span id="id2">[]</span> - AlphaFold structure prediction using embeddings</p></li>
<li><p><span id="id3">[]</span> - Molecular embeddings for drug design</p></li>
<li><p><span id="id4">[]</span> - Structure-aware protein embeddings</p></li>
</ul>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./content/ai_pi_book/part2"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="part2_why_modern_ai_feels_alien.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Why Modern AI Feels Alien (Even for researchers With Statistical and ML Background)</p>
      </div>
    </a>
    <a class="right-next"
       href="part2_chapter2_multi_task_learning.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Multi-Task Learning: One Model, Many Outputs</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-fundamental-paradigm-shift">The Fundamental Paradigm Shift</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#what-are-embeddings">What Are Embeddings?</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#key-properties-of-good-embeddings">Key Properties of Good Embeddings</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#from-hand-crafted-features-to-learned-representations">From Hand-Crafted Features to Learned Representations</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#the-traditional-approach">The Traditional Approach</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#the-embedding-revolution">The Embedding Revolution</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#how-embeddings-are-learned">How Embeddings Are Learned</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#self-supervised-learning">Self-Supervised Learning</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#contrastive-learning">Contrastive Learning</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#why-embeddings-matter-for-biology">Why Embeddings Matter for Biology</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#transfer-learning">1. Transfer Learning</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#multi-task-learning">2. Multi-Task Learning</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#similarity-search-and-clustering">3. Similarity Search and Clustering</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#interpretable-latent-spaces">4. Interpretable Latent Spaces</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#practical-considerations">Practical Considerations</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#embedding-quality-depends-on-training-data">Embedding Quality Depends on Training Data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#not-all-layers-are-equal">Not All Layers Are Equal</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#dimensionality-matters">Dimensionality Matters</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#common-pitfalls">Common Pitfalls</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#key-takeaways">Key Takeaways</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#further-reading">Further Reading</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Babu group
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>